packages/horizon/contracts/payments/GraphPayments.sol

```diff
@@ -69,7 +69,11 @@ contract GraphPayments is Multicall, GraphDirectory, GraphPaymentsStorageV1Stora
         }
 
         // Pay the rest to the receiver
-        uint256 tokensReceiverRemaining = tokens - tokensProtocol - tokensDataService - tokensDelegationPool;
+        uint256 totalCut = tokensProtocol + tokensDataService + tokensDelegationPool;
+        if (totalCut > tokens) {
+            revert GraphPaymentsCollectorInsufficientAmount(tokens, totalCut);
```

Comment:
Not sure I would use the term `Collector` here, how about just:
```suggestion
            revert GraphPaymentsInsufficientTokens(tokens, totalCut);
```

Reply:
totally!